tensorflow/contrib/metrics/python/kernel_tests/confusion_matrix_ops_test.py

```diff
@@ -0,0 +1,127 @@
+    with self.test_session():
+      ans = tf.contrib.metrics.confusion_matrix(predictions, targets)
+      tf_ans = ans.eval()
+      self.assertAllClose(tf_ans, truth, atol=1e-10)
+
+  def _testBasic(self, dtype):
+    predictions = np.arange(5, dtype=dtype)
+    targets = np.arange(5, dtype=dtype)
+
+    truth = np.asarray(
+      [[1, 0, 0, 0, 0],
+       [0, 1, 0, 0, 0],
+       [0, 0, 1, 0, 0],
+       [0, 0, 0, 1, 0],
+       [0, 0, 0, 0, 1]],
+      dtype=dtype)
+
+    self._testConfMatrix(
+      predictions=predictions,
+      targets=targets,
+      truth=truth)
+
+  def testInt32Basic(self, dtype=np.int32):
+    self._testBasic(dtype)
+
+  def testInt64Basic(self, dtype=np.int64):
+    self._testBasic(dtype)
+
+  def _testDiffentLabelsInPredictionAndTarget(self, dtype):
+    predictions = np.asarray([1, 2, 3], dtype=dtype)
+    targets = np.asarray([4, 5, 6], dtype=dtype)
+
+    truth = np.asarray(
+      [[0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 1, 0, 0],
+       [0, 0, 0, 0, 0, 1, 0],
+       [0, 0, 0, 0, 0, 0, 1],
+       [0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 0]],
+      dtype=dtype)
+
+    self._testConfMatrix(
+      predictions=predictions,
+      targets=targets,
+      truth=truth)
+
+  def testInt32DifferentLabels(self, dtype=np.int32):
+    self._testDiffentLabelsInPredictionAndTarget(dtype)
+
+  def testInt64DifferentLabels(self, dtype=np.int64):
+    self._testDiffentLabelsInPredictionAndTarget(dtype)
+
+  def _testMultipleLabels(self, dtype):
+    predictions = np.asarray([1, 1, 2, 3, 5, 6, 1, 2, 3, 4], dtype=dtype)
+    targets = np.asarray([1, 1, 2, 3, 5, 1, 3, 6, 3, 1], dtype=dtype)
+
+    truth = np.asarray(
+      [[0, 0, 0, 0, 0, 0, 0],
+       [0, 2, 0, 1, 0, 0, 0],
+       [0, 0, 1, 0, 0, 0, 1],
+       [0, 0, 0, 2, 0, 0, 0],
+       [0, 1, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 1, 0],
+       [0, 1, 0, 0, 0, 0, 0]],
+      dtype=dtype)
+
+    self._testConfMatrix(
+      predictions=predictions,
+      targets=targets,
+      truth=truth)
+
+  def testInt32MultipleLabels(self, dtype=np.int32):
+    self._testMultipleLabels(dtype)
+
+  def testInt64MultipleLabels(self, dtype=np.int64):
+    self._testMultipleLabels(dtype)
+
+  def testInvalidRank(self):
+    predictions = np.asarray([[1, 2, 3]])
+    targets = np.asarray([1, 2, 3])
+    self.assertRaisesRegexp(
+        ValueError, "are not compatible",
+        tf.contrib.metrics.confusion_matrix, predictions, targets)
+
+    predictions = np.asarray([1, 2, 3])
+    targets = np.asarray([[1, 2, 3]])
+    self.assertRaisesRegexp(
+        ValueError, "are not compatible",
+        tf.contrib.metrics.confusion_matrix, predictions, targets)
+
+  def testInputDifferentSize(self):
+    predictions = np.asarray([1, 2, 3])
+    targets = np.asarray([1, 2])
+    self.assertRaisesRegexp(
+          ValueError,
+          "are not compatible",
+          tf.contrib.metrics.confusion_matrix, predictions, targets)
```

Comment:
I have rerun both tests and both of them are passing. Maybe another operation is raising the exceptions. But if it is better, I can remove or rewrite them.